code.js

```diff
@@ -1,3 +1,46 @@
-function dijkstra(graph, sourceNode) {
-    return [];
+class PriorityQueue {
+    constructor() {
+        this.items = [];
+    }
+    enqueue(element, priority) {
+        this.items.push({ element, priority });
+        this.items.sort((a, b) => a.priority - b.priority); // Sort by priority (ascending)
```

Comment:
You don't need to sort the entire list again when enqueuing a new element -- this is very inefficient. How can you do it better?